File: apps/furniture-store-service-server/src/APIs/Product/Dtos/ProductWhereInput.cs
namespace FurnitureStoreService.APIs.Dtos;

public class ProductWhereInput
{
    public string? Id { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public double? Price { get; set; }

    public string? Category { get; set; }

    public List<string>? OrderItems { get; set; }
}

File: apps/my-service-admin/src/api/product/ProductWhereInput.ts
import { StringFilter } from "../../util/StringFilter";
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { FloatNullableFilter } from "../../util/FloatNullableFilter";
import { CategoryWhereUniqueInput } from "../category/CategoryWhereUniqueInput";

export type ProductWhereInput = {
  id?: StringFilter;
  name?: StringNullableFilter;
  description?: StringNullableFilter;
  price?: FloatNullableFilter;
  sku?: StringNullableFilter;
  category?: CategoryWhereUniqueInput;
};
